L03-generics/src/main/java/ru/otus/homework/CustomerService.java

```diff
@@ -0,0 +1,54 @@
+package ru.otus.homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    private Comparator<Customer> customerComparator = Comparator.comparing(Customer::getScores);
+    private NavigableMap<Customer, String> customerData = new TreeMap<>(customerComparator);
+
+    public Map.Entry<Customer, String> getSmallest() {
+        return customerData.firstEntry() == null
+                ? null
+                : Map.entry(Customer.copy(customerData.firstEntry().getKey()), customerData.firstEntry().getValue());
+    }
+
+    public Map.Entry<Customer, String> getNext(Customer customer) {
+        if (!isSorted(customerData)) {
+            customerData = copy(customerData);
+        }
+
+        return customer == null
+                ? null
+                : customerData.ceilingEntry(customer);
```

Comment:
```suggestion
                : customerData.higherEntry(customer);
```

Здесь надо строго больше (`higherEntry`), хотя тестов на "равно" нет и с `ceilingEntry` тоже работает.